File: AtlasWeb/Core/Modules/Tibbi_Surec/Onemli_Tibbi_Bilgiler/MedicalInformationFormViewModel.ts
//$523E5A93
import { BaseViewModel } from "NebulaClient/Model/BaseViewModel";
import { MedicalInformation } from 'NebulaClient/Model/AtlasClientModel';
import { MedicalInfoFoodAllergies } from 'NebulaClient/Model/AtlasClientModel';
import { MedicalInfoDrugAllergies } from 'NebulaClient/Model/AtlasClientModel';
import { MedicalInfoDisabledGroup } from 'NebulaClient/Model/AtlasClientModel';
import { MedicalInfoHabits } from 'NebulaClient/Model/AtlasClientModel';
import { SKRSSigaraKullanimi } from 'NebulaClient/Model/AtlasClientModel';
import { SKRSAlkolKullanimi } from 'NebulaClient/Model/AtlasClientModel';
import { MedicalInfoAllergies } from 'NebulaClient/Model/AtlasClientModel';
import { DietMaterialDefinition } from 'NebulaClient/Model/AtlasClientModel';
import { ActiveIngredientDefinition } from 'NebulaClient/Model/AtlasClientModel';
import { Type } from 'NebulaClient/ClassTransformer';

export class MedicalInformationFormViewModel extends BaseViewModel {
    @Type(() => MedicalInformation)
    public _MedicalInformation: MedicalInformation = new MedicalInformation();
    @Type(() => MedicalInfoFoodAllergies)
    public MedicalInfoFoodAllergiesMedicalInfoFoodAllergiesGridList: Array<MedicalInfoFoodAllergies> = new Array<MedicalInfoFoodAllergies>();
    @Type(() => MedicalInfoDrugAllergies)
    public MedicalInfoDrugAllergiesMedicalInfoDrugAllergiesGridList: Array<MedicalInfoDrugAllergies> = new Array<MedicalInfoDrugAllergies>();
    @Type(() => MedicalInfoDisabledGroup)
    public MedicalInfoDisabledGroups: Array<MedicalInfoDisabledGroup> = new Array<MedicalInfoDisabledGroup>();
    @Type(() => MedicalInfoHabits)
    public MedicalInfoHabitss: Array<MedicalInfoHabits> = new Array<MedicalInfoHabits>();
    @Type(() => SKRSSigaraKullanimi)
    public SKRSSigaraKullanimis: Array<SKRSSigaraKullanimi> = new Array<SKRSSigaraKullanimi>();
    @Type(() => SKRSAlkolKullanimi)
    public SKRSAlkolKullanimis: Array<SKRSAlkolKullanimi> = new Array<SKRSAlkolKullanimi>();
    @Type(() => MedicalInfoAllergies)
    public MedicalInfoAllergiess: Array<MedicalInfoAllergies> = new Array<MedicalInfoAllergies>();
    @Type(() => DietMaterialDefinition)
    public DietMaterialDefinitions: Array<DietMaterialDefinition> = new Array<DietMaterialDefinition>();
    @Type(() => ActiveIngredientDefinition)
    public ActiveIngredientDefinitions: Array<ActiveIngredientDefinition> = new Array<ActiveIngredientDefinition>();
    @Type(() => Object)
    public _selectedMedicalInfoFoodAllergyValue: Object;
    @Type(() => Object)
    public _selectedMedicalInfoDrugAllergyValue: Object;
    public _PatientID: string;    
    public checkBoxControl : boolean = true;
    public buttonControl: boolean = true;
    public allergyControl: boolean = true;
    public changeHighRiskPregnancy: boolean = false;
}

File: AtlasWeb/Core/Modules/Tibbi_Surec/Onemli_Tibbi_Bilgiler/MedicalInformationFormViewModel.cs
//$5F684092
using System;
using System.Linq;
using System.Net.Http;
using System.ComponentModel;
using TTInstanceManagement;
using Core.Models;
using TTObjectClasses;

using Infrastructure.Filters;
using TTDefinitionManagement;
using Microsoft.AspNetCore.Mvc;

namespace Core.Controllers
{
    public partial class MedicalInformationServiceController : Controller
    {
        partial void PreScript_MedicalInformationForm(MedicalInformationFormViewModel viewModel, MedicalInformation medicalInformation, TTObjectContext objectContext)
        {
            if (medicalInformation.MedicalInfoAllergies == null)
            {
                medicalInformation.MedicalInfoAllergies = new MedicalInfoAllergies(objectContext);
            }
            if (medicalInformation.MedicalInfoHabits == null)
            {
                medicalInformation.MedicalInfoHabits = new MedicalInfoHabits(objectContext);
            }
            if (medicalInformation.MedicalInfoDisabledGroup == null)
            {
                medicalInformation.MedicalInfoDisabledGroup = new MedicalInfoDisabledGroup(objectContext);
            }

            ContextToViewModel(viewModel, objectContext);

        }

        partial void PostScript_MedicalInformationForm(MedicalInformationFormViewModel viewModel, MedicalInformation medicalInformation, TTDefinitionManagement.TTObjectStateTransitionDef transDef, TTObjectContext objectContext)
        {
            if (medicalInformation.Patient == null || medicalInformation.Patient.Count() == 0)
            {
                Guid? activePatientObjectID = viewModel.GetSelectedPatientID();
                if (activePatientObjectID.HasValue)
                {
                    Patient activepatient = objectContext.GetObject<Patient>(activePatientObjectID.Value);
                    activepatient.MedicalInformation = medicalInformation;
                }
            }

            if(medicalInformation.HighRiskPregnancy != null && viewModel.changeHighRiskPregnancy == true)
            {
                medicalInformation.HighRiskPregnancyDate = Common.RecTime();
            }
        }
        partial void AfterContextSaveScript_MedicalInformationForm(MedicalInformationFormViewModel viewModel, MedicalInformation medicalInformation, TTDefinitionManagement.TTObjectStateTransitionDef transDef, TTObjectContext objectContext)
        {


        }

    }
}

namespace Core.Models
{
    public partial class MedicalInformationFormViewModel
    {
        //public TTObjectClasses.MedicalInformation _MedicalInformation { get; set; }
        //public TTObjectClasses.MedicalInfoFoodAllergies[] MedicalInfoFoodAllergiesMedicalInfoFoodAllergiesGridList { get; set; }
        //public TTObjectClasses.MedicalInfoDrugAllergies[] MedicalInfoDrugAllergiesMedicalInfoDrugAllergiesGridList { get; set; }
        //public TTObjectClasses.MedicalInfoDisabledGroup[] MedicalInfoDisabledGroups { get; set; }
        //public TTObjectClasses.MedicalInfoHabits[] MedicalInfoHabitss { get; set; }
        //public TTObjectClasses.SKRSSigaraKullanimi[] SKRSSigaraKullanimis { get; set; }
        //public TTObjectClasses.SKRSAlkolKullanimi[] SKRSAlkolKullanimis { get; set; }
        //public TTObjectClasses.MedicalInfoAllergies[] MedicalInfoAllergiess { get; set; }
        //public TTObjectClasses.DietMaterialDefinition[] DietMaterialDefinitions { get; set; }
        //public TTObjectClasses.ActiveIngredientDefinition[] ActiveIngredientDefinitions { get; set; }
        public string _PatientID
        {
            get;
            set;
        }

        public object _selectedMedicalInfoFoodAllergyValue
        {
            get;
            set;
        }

        public object _selectedMedicalInfoDrugAllergyValue
        {
            get;
            set;
        }
        public bool checkBoxControl
        {
            get;
            set;
        }
        public bool buttonControl { get; set; }

        public bool allergyControl { get; set; }

        public bool changeHighRiskPregnancy { get; set; }
    }
}